openmls/tests/key_store.rs

```diff
@@ -0,0 +1,82 @@
+//! A couple of simple tests on how to interact with the key store.
+use openmls::{prelude::*, test_utils::*, *};
+use openmls_traits::{key_store::OpenMlsKeyStore, types::SignatureScheme};
+
+#[apply(ciphersuites_and_backends)]
+fn test_store_key_package_bundle(
+    ciphersuite: &'static Ciphersuite,
+    backend: &impl OpenMlsCryptoProvider,
+) {
+    // First we generate a credential and key package for our user.
+    let credential_bundle = CredentialBundle::new(
+        b"User ID".to_vec(),
+        CredentialType::Basic,
+        SignatureScheme::from(ciphersuite.name()),
+        backend,
+    )
+    .unwrap();
+    let key_package_bundle =
+        KeyPackageBundle::new(&[ciphersuite.name()], &credential_bundle, backend, vec![])
+            .expect("Error generating new key package bundle.");
+
+    // In order to store something in the key store we need to define an ID.
+    // Here we simply take the key package reference.
+    let id = key_package_bundle
+        .key_package()
+        .hash_ref(backend.crypto())
+        .expect("Failed to hash KeyPackage.");
+
+    // Now we can store the key_package_bundle.
+    backend
+        .key_store()
+        .store(id.as_slice(), &key_package_bundle)
+        .expect("Failed to store key package bundle in keystore.");
+
+    // Delete the key package bundle.
+    backend
+        .key_store()
+        .delete(id.as_slice())
+        .expect("Error deleting key package bundle");
+}
+
+#[apply(ciphersuites_and_backends)]
+fn test_read_credential_bundle(
+    ciphersuite: &'static Ciphersuite,
+    backend: &impl OpenMlsCryptoProvider,
+) {
+    // First we generate a credential bundle
+    let credential_bundle_to_store = CredentialBundle::new(
+        b"User ID".to_vec(),
+        CredentialType::Basic,
+        SignatureScheme::from(ciphersuite.name()),
+        backend,
+    )
+    .unwrap();
+
+    let credential = credential_bundle_to_store.credential();
+
+    let id = credential
+        .signature_key()
+        .tls_serialize_detached()
+        .expect("Error serializing the credential's public key.");
+
+    // Now we can store the credential_bundle.
+    backend
+        .key_store()
+        .store(id.as_slice(), &credential_bundle_to_store)
+        .expect("Failed to store credential in keystore.");
+
+    // In order to read something from the key store we need to define an ID.
+    // Here we simply take the serialized public key of the credential.
```

Comment:
This comment should probably go in line 57.

Reply:
nope, 57 is not in the book (it's just setup)